File: src/main/java/br/com/useinet/finance/controller/ContaController.java
package br.com.useinet.finance.controller;

import br.com.useinet.finance.dto.ContaRequest;
import br.com.useinet.finance.dto.ContaResponse;
import br.com.useinet.finance.model.Usuario;
import br.com.useinet.finance.service.ContaService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.security.SecurityRequirement;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/contas")
@Tag(name = "Contas", description = "Gerenciamento de contas bancárias do usuário")
@SecurityRequirement(name = "Bearer Authentication")
public class ContaController {

    private final ContaService contaService;

    public ContaController(ContaService contaService) {
        this.contaService = contaService;
    }

    @GetMapping
    @Operation(summary = "Listar contas do usuário")
    public ResponseEntity<List<ContaResponse>> listar(@AuthenticationPrincipal Usuario usuario) {
        return ResponseEntity.ok(contaService.listar(usuario));
    }

    @PostMapping
    @Operation(summary = "Criar conta")
    public ResponseEntity<ContaResponse> criar(@RequestBody ContaRequest request,
                                               @AuthenticationPrincipal Usuario usuario) {
        return ResponseEntity.status(HttpStatus.CREATED).body(contaService.criar(request, usuario));
    }

    @PutMapping("/{id}")
    @Operation(summary = "Atualizar conta")
    public ResponseEntity<ContaResponse> atualizar(@PathVariable Long id,
                                                   @RequestBody ContaRequest request,
                                                   @AuthenticationPrincipal Usuario usuario) {
        return ResponseEntity.ok(contaService.atualizar(id, request, usuario));
    }

    @DeleteMapping("/{id}")
    @Operation(summary = "Excluir conta")
    public ResponseEntity<Void> deletar(@PathVariable Long id,
                                        @AuthenticationPrincipal Usuario usuario) {
        contaService.deletar(id, usuario);
        return ResponseEntity.noContent().build();
    }
}

File: src/main/kotlin/br/com/useinet/finance/controller/ContaController.kt
package br.com.useinet.finance.controller

import br.com.useinet.finance.dto.ContaRequest
import br.com.useinet.finance.dto.ContaResponse
import br.com.useinet.finance.model.Usuario
import br.com.useinet.finance.service.ContaService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.security.SecurityRequirement
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/contas")
@Tag(name = "Contas", description = "Gerenciamento de contas bancárias do usuário")
@SecurityRequirement(name = "Bearer Authentication")
class ContaController(private val contaService: ContaService) {

    @GetMapping
    @Operation(summary = "Listar contas do usuário")
    fun listar(@AuthenticationPrincipal usuario: Usuario): ResponseEntity<List<ContaResponse>> =
        ResponseEntity.ok(contaService.listar(usuario))

    @PostMapping
    @Operation(summary = "Criar conta")
    fun criar(
        @RequestBody request: ContaRequest,
        @AuthenticationPrincipal usuario: Usuario
    ): ResponseEntity<ContaResponse> =
        ResponseEntity.status(HttpStatus.CREATED).body(contaService.criar(request, usuario))

    @PutMapping("/{id}")
    @Operation(summary = "Atualizar conta")
    fun atualizar(
        @PathVariable id: Long,
        @RequestBody request: ContaRequest,
        @AuthenticationPrincipal usuario: Usuario
    ): ResponseEntity<ContaResponse> =
        ResponseEntity.ok(contaService.atualizar(id, request, usuario))

    @DeleteMapping("/{id}")
    @Operation(summary = "Excluir conta")
    fun deletar(@PathVariable id: Long, @AuthenticationPrincipal usuario: Usuario): ResponseEntity<Void> {
        contaService.deletar(id, usuario)
        return ResponseEntity.noContent().build()
    }
}
